web/srv/api_handlers.go

```diff
@@ -168,10 +170,34 @@ func (h *handler) handleAPITopRoutes(w http.ResponseWriter, req *http.Request, p
 	renderJSONPb(w, result)
 }
 
+// Control frame payload size must be no longer than 123 bytes. In the case of
+// an unexpected HTTP status code or unexpected error, truncate the message
+// after 123 bytes so that the web socket message is properly written.
+func createTapErrorMessage(err error) string {
+	var msg string
+
+	if httpErr, ok := err.(protohttp.HTTPError); ok {
+		log.Debugf("tap error is HTTPError: %s", httpErr.Error())
+		if httpErr.Code == http.StatusForbidden {
+			msg = fmt.Sprintf("tap authorization failed, visit %s for more information", tap.TapRbacURL)
+		} else {
+			msg = httpErr.WrappedError.Error()[:123]
+		}
+	} else {
+		log.Debugf("tap error: %s", err.Error())
+		msg = err.Error()[:123]
+	}
+
+	return msg
```

Comment:
i think it's useful to return the code from non-StatusForbidden `HTTPError`'s, rather than just the `Wrapped` error. you can treat non-StatusForbidden `HTTPError`'s and all other non-HTTPError's the same:

```go
// websocket.maxControlFramePayloadSize - 2 bytes for payload size
const maxControlFrameMsgSize = 123

func createTapErrorMessage(err error) string {
	log.Debugf("tap error: %s", err)

	if httpErr, ok := err.(protohttp.HTTPError); ok && httpErr.Code == http.StatusForbidden {
		return fmt.Sprintf("tap authorization failed, visit %s for more information", tap.TapRbacURL)
	}

	msg := err.Error()
	if len(msg) > maxControlFrameMsgSize {
		return msg[:maxControlFrameMsgSize]
	}

	return msg
}
```

Reply:
Good catch! I'll make this change and the others above.